internal/client/upbound.go

```diff
@@ -46,36 +46,37 @@ var (
 	DefaultAPIEndpoint, _ = url.Parse("https://api.upbound.io")
 )
 
-func NewConfig(ctx context.Context, kube client.Client, mg resource.Managed) (*up.Config, error) {
+func NewConfig(ctx context.Context, kube client.Client, mg resource.Managed) (*up.Config, Profile, error) {
 	pc := &v1alpha1.ProviderConfig{}
+	profile := Profile{}
 	if err := kube.Get(ctx, types.NamespacedName{Name: mg.GetProviderConfigReference().Name}, pc); err != nil {
-		return nil, errors.Wrapf(err, "cannot get provider config %s", mg.GetProviderConfigReference().Name)
+		return nil, profile, errors.Wrapf(err, "cannot get provider config %s", mg.GetProviderConfigReference().Name)
 	}
 
 	data, err := resource.CommonCredentialExtractor(ctx, pc.Spec.Credentials.Source, kube, pc.Spec.Credentials.CommonCredentialSelectors)
 	if err != nil {
-		return nil, errors.Wrap(err, "cannot get credentials")
+		return nil, profile, errors.Wrap(err, "cannot get credentials")
 	}
 	cliConfig := &CLIConfig{}
 	if err := json.Unmarshal(data, cliConfig); err != nil {
-		return nil, errors.Wrap(err, "cannot unmarshal credentials")
+		return nil, profile, errors.Wrap(err, "cannot unmarshal credentials")
 	}
-	profile := cliConfig.Upbound.Profiles[cliConfig.Upbound.Default]
+	profile = cliConfig.Upbound.Profiles[cliConfig.Upbound.Default]
 	if len(profile.Session) == 0 {
-		return nil, errors.New("no session found")
+		return nil, profile, errors.New("no session found")
```

Comment:
```suggestion
		return nil, Profile{}, errors.New("no session found")
```

In case of errors, it's better to not return any potentially non-empty information.